src/appleseed/renderer/kernel/intersection/embreescene.cpp

```diff
@@ -0,0 +1,518 @@
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Fedor Matantsev, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "embreescene.h"
+
+// Standard headers.
+#include <memory>
+
+// Embree headers.
+#include <embree3/rtcore.h>
+
+// Foundation headers.
+#include "foundation/math/area.h"
+#include "foundation/math/fp.h"
+#include "foundation/math/intersection/raytrianglemt.h"
+#include "foundation/math/scalar.h"
+#include "foundation/math/transform.h"
+#include "foundation/utility/makevector.h"
+#include "foundation/utility/statistics.h"
+#include "foundation/utility/stopwatch.h"
+
+// appleseed.renderer headers.
+#include "renderer/kernel/intersection/intersectionsettings.h"
+#include "renderer/kernel/shading/shadingpoint.h"
+#include "renderer/kernel/shading/shadingray.h"
+#include "renderer/modeling/object/curveobject.h"
+#include "renderer/modeling/object/iregion.h"
+#include "renderer/modeling/object/meshobject.h"
+#include "renderer/modeling/object/object.h"
+#include "renderer/modeling/object/triangle.h"
+#include "renderer/modeling/scene/assembly.h"
+#include "renderer/modeling/scene/containers.h"
+
+using namespace foundation;
+using namespace std;
+
+
+namespace
+{
+
+using namespace renderer;
```

Comment:
Indent comment of anonymous namespaces.

The rule is: we always indent the content of namespaces. The exception is: we don't indent the content of _top level_ namespaces since that would amount to indenting the whole file which doesn't bring anything to the table.

Reply:
Move `using namespace renderer` clause at the translation unit level, next to the other ones.